packages/client/src/links/internals/httpUtils.ts

```diff
@@ -194,11 +196,14 @@ export function httpRequest(
         });
       })
       .catch((err) => {
+        done = true;
         reject(TRPCClientError.from(err, { meta }));
       });
```

Comment:
You can also use `.finally(() => { done = true; });` to have everything in one place.

Reply:
Unfortunately I can't because calling `reject()`/`resolve()` causes a race condition that gets to the cleanup function before `.finally()` is executed